### ISSUE TITLE
OpenedX: configuration and installation of ol_openedx_checkout_external

### ISSUE THREAD

**Issue by rhysyngsun** [mitodl/mitxonline#567]
This work can be done by anyone on the team even though it's infrastructure. Steps to perform this bellow:

- [ ] Add an entry to install the plugin to each of the environments [here](https://github.com/mitodl/ol-infrastructure/blob/main/src/bilder/images/edxapp/group_data/all.py#L23-L73)
- [ ] Add environment variables to the [OpenedX settings template](https://github.com/mitodl/ol-infrastructure/blob/8ff401b29776cb226a3aa7c943f5db75cb4cce28/src/bilder/images/edxapp/templates/edxapp/mitxonline/common_values.yml.tmpl):
  - `MARKETING_SITE_CHECKOUT_URL`
  - `ECOMMERCE_PUBLIC_URL_ROOT
- Coordinate with devops creating:
  - Create an new ecommerce configuration in http://<LMS_BASE>/admin/commerce/commerceconfiguration with "basket_checkout_page=/checkout-external/"`

**Comment by arslanashraf7** [comment]
As a next step of the ticket I've setup a course and configurations in the Django admin to test this ticket in QA/RC. The steps below might be referred for creating future courses for the external checkout.

- I've set up a course for testing https://rc.mitxonline.mit.edu/courses/course-v1:MITx+CHECKOUT/ (In MITxOnline RC)
- The mapped course in MITxOnline's edx instance is https://courses-qa.mitxonline.mit.edu/learn/course/course-v1:Arbisoft+ARB_TEST_CKT_001+1/home
- I've added the connfiguration for ecommerce in admin at https://courses-qa.mitxonline.mit.edu/admin/commerce/commerceconfiguration/1
- I've added course modes for this test course at https://courses-qa.mitxonline.mit.edu/admin/course_modes/coursemode/

While setting up a course in edX and enabling it to show the `Upgrade` button we need to make a couple of tweaks, mentioning them below:

- There should be both `Audit` & `Verified` course modes for the course
- User should initially be enrolled in `Audit` mode 
- The `Verified` mode of the course should have an upgrade deadline set

Once the above steps are complete the user would see the upgrade button in edX and when the `external-checkout` plugin is set up the user should be redirected to MITxOnline's cart.

FYI: @briangrossman @pdpinch @rhysyngsun @blarghmatey

**Comment by arslanashraf7** [comment]
@blarghmatey I think the `ECOMMERCE_PUBLIC_URL_ROOT` value is not picked up properly in our MITxOnline QA instance. The value was set in [here](https://github.com/mitodl/ol-infrastructure/pull/823/files#diff-dc8dde90c09221240c790ee762d8332b646c5d580906faa3a390a9aba7c853bfR526) in a recent PR. But when we open [this test course](https://courses-qa.mitxonline.mit.edu/learn/course/course-v1:Arbisoft+ARB_TEST_CKT_001+1/home) and navigate to `Upgrade` button it tries to take the user to something like `http://localhost:8002/checkout-external/?sku=TESTSKU` whereas the base URL should be same as the LMS base.

Could you tell what could be going wrong here?

**Comment by blarghmatey** [comment]
The new instance hadn't been deployed yet so the configs weren't live yet. I just ran the deployment so this should be working now.

**Comment by arslanashraf7** [comment]
> The new instance hadn't been deployed yet so the configs weren't live yet. I just ran the deployment so this should be working now.

@blarghmatey Thanks for the new build. 

Now the eCommerce URL is correct but somehow [MARKETING_SITE_CHECKOUT_URL setting value](https://github.com/mitodl/ol-infrastructure/commit/78cdb19206bcc816c0f859cd817672eeb9d91220#diff-dc8dde90c09221240c790ee762d8332b646c5d580906faa3a390a9aba7c853bfR337) is not getting picked up properly and the plugin throws [this configuration error](https://sentry.io/organizations/mit-office-of-digital-learning/issues/3289913963/?project=5939801&query=is%3Aunresolved). Could you tell why this value might not be getting picked?

**Comment by blarghmatey** [comment]
I'm not sure what the disconnect is with that configuration value. I logged into a running instance and verified that the setting is present in `/edx/etc/lms.yml` and that the `ol-openedx-checkout-external` plugin is installed at version 0.1.0

Let me know if you would like me to do further investigation or if it's something that you would like to look into at the plugin level.

**Comment by rhysyngsun** [comment]
@arslanashraf7 I think maybe the plugin needs something like this to load the vars from the env like this other plugin does:

https://github.com/mitodl/open-edx-plugins/blob/76ef69de89669d1c8a534192c767e68b02e318a4/src/ol_openedx_sentry/settings/sentry.py#L102-L104

I'm not quite sure how this worked locally for me, but based on the current source code `settings.MARKETING_SITE_CHECKOUT_URL` is always `None`.

@blarghmatey does that seem right? I'm not super familiar with the `ENV_TOKENS` stuff used there.

**Comment by blarghmatey** [comment]
Yeah, that seems likely. I know that it doesn't automatically inherit from the edX django settings, so it will probably need to pull the values from that function interface and then use it from there.

**Comment by arslanashraf7** [comment]
@rhysyngsun Yes, I figured the `plugin_settings` method does load the settings but the problem in our case was that we are not loading them from the settings passed in `production.py` which we should. I was also looking at [Canvas plugin implementation](https://github.com/mitodl/open-edx-plugins/blob/76ef69de89669d1c8a534192c767e68b02e318a4/src/ol_openedx_canvas_integration/settings/production.py#L8-L15) which also adds some settings and they are working fine. So the problem is in the settings in `production.py` in checkout plugin which also sets defult to None. I'll create a PR to update them in checkout plugin.

**Comment by rhysyngsun** [comment]
@arslanashraf7 @blarghmatey are we all set here? RC looks ok at least (and I wouldn't expect to see it in prod yet)

**Comment by arslanashraf7** [comment]
The infra-related change was done in [this PR](https://github.com/mitodl/ol-infrastructure/pull/823/files) e.g. adding the plugin + configurations. Also, there were specific details [in this slack thread](https://mitodl.slack.com/archives/C02649X2P1V/p1654601128972659?thread_ts=1654549259.504609&cid=C02649X2P1V) about building the AMIs that transition from CI->QA->Prod.  

I think keeping it just on RC even if we deploy a new AMI build for prod @blarghmatey might guide better.
